packit_service/worker/build/build_helper.py

```diff
@@ -91,6 +95,35 @@ def local_project(self) -> LocalProject:
             )
         return self._local_project
 
+    @property
+    def pr_id(self) -> int:
+        if self._pr_id is None:
+            self._pr_id = int(self.metadata.event_dict["identifier"])
+        return self._pr_id
+
+    @property
+    def check_project_permission(self) -> bool:
+        username = self.project.service.user.get_username()
+        if self._check_project_permission is None:
+            self._check_project_permission = self.base_project.can_merge_pr(username)
+        return self._check_project_permission
+
+    @property
+    def base_project(self) -> GitProject:
+        """
+        Getting the source project info from PR,
+        In case of build events we loose the source info.
+        """
+        if self._base_project is None:
+            self._base_project = self.project.get_pr(pr_id=self.pr_id).source_project
+        return self._base_project
+
+    def request_project_access(self) -> None:
+        try:
+            self.base_project.request_access()
+        except Exception:
```

Comment:
I would use the most specific exception possible and ideally check also content. We can get more exception here. (E.g. network/authentication problems...).

Reply:
Yes added more specific exceptions.